utils/logging.js

```diff
@@ -0,0 +1,30 @@
+// @ts-check
+const morgan = require("morgan");
+
+morgan.token("locals", (req, res) => {
+  if (Object.keys(res.locals).length) {
+    return JSON.stringify(res.locals);
+  }
+});
+
+const format =
+  ":date[iso] | :remote-addr | :method :status :url | :referrer | :res[content-length] | :response-time ms | :locals";
+
+/** @type {import('morgan').Options} */
+const options = {
+  skip(req, res) {
+    const { method, path, hostname } = req;
+    const { statusCode } = res;
+    switch (true) {
```

Comment:
This is the weirdest thing I’ve seen in a while :) maybe just `if`s?

Reply:
the conditions in if were getting nastily long.. so decided to use this. will move back to regular code 😅

Reply:
doesn't look so bad now